### PR TITLE
Refactor:  Header -> Signed Header

### PR DESCRIPTION
**Pull request by nashqueue**
Fixes: #773

### CODE REVIEW

File: proto/rollkit/rollkit.proto [hunk 1]
```diff
@@ -49,19 +49,19 @@ message Header {
...
 
 message Commit {
-	uint64 height = 1;
-	bytes header_hash = 2;
-	// Note: most of the time this will be a single sinature
-	repeated bytes signatures = 3;
+	repeated bytes signatures = 1;
 }
 
 message SignedHeader {
```
Comment (gupadhyaya) [inline proto/rollkit/rollkit.proto:67]: i was wondering if we don't modify the `Header` with additional fields like public key, but just add signature and public key directly to `SignedHeader`, just like the way celestia-node does it for their `ExtendedHeader`?
```
message ExtendedHeader {
  tendermint.types.Header header = 1;
  tendermint.types.Commit commit = 2;
  tendermint.types.ValidatorSet validator_set = 3;
}
```
similarly, we could do
```
message SignedHeader {
   Header header = 1;
   repeated bytes signatures = 2;
   repeated bytes signer_public_keys = 3;
}
```
or we could retain the tendermint types `Commit` and `ValidatorSet`
```
message SignedHeader {
   Header header = 1;
   tendermint.types.Commit commit = 2;
   tendermint.types.ValidatorSet validator_set = 3;
}
```
retaining tendermint types would help with less internal conversions.

Comment (nashqueue) [reply proto/rollkit/rollkit.proto:67]: I like the proposal. 2 Things i would like to add: 

1. We should try not to add tendermint types if we can keep them out. 

2. The `repeated bytes signer_public_keys = 3` must be inside the header as we must sign over the validator set changes.

Comment (gupadhyaya) [reply proto/rollkit/rollkit.proto:67]: > 2\. The `repeated bytes signer_public_keys = 3` must be inside the header as we must sign over the validator set changes.

there is already `AggregatorsHash` part of the header which being signed and it is computed as hash of the validatorset.

Comment (nashqueue) [reply proto/rollkit/rollkit.proto:67]: I propose we remove the hash and add the validator pub keys to the header. We are gossiping about everything anyways, so I don't see an overhead here. There are definitely improvements /optimizations possible that we could follow up on later.

File: proto/rollkit/rollkit.proto [hunk 2]
```diff
@@ -49,19 +49,19 @@ message Header {
...
-	string chain_id = 12;
+	string chain_id = 13;
 }
 
 message Commit {
-	uint64 height = 1;
-	bytes header_hash = 2;
-	// Note: most of the time this will be a single sinature
-	repeated bytes signatures = 3;
+	repeated bytes signatures = 1;
```
Comment (gupadhyaya) [inline proto/rollkit/rollkit.proto:64]: you can move `repeated bytes signatures` to `SignedHeader` as it is the only field in the `Commit`

**Review by gupadhyaya — DISMISSED** [review]

File: proto/rollkit/rollkit.proto [hunk 3]
```diff
@@ -49,19 +49,19 @@ message Header {
 	// pubkey can't be recovered by the signature (e.g. ed25519).
 	bytes proposer_address = 10;
 
+	//public key of the proposer so we can verify the signature
```
Comment (tuxcanfly) [inline proto/rollkit/rollkit.proto:52]: nit: spacing at comment start

File: proto/rollkit/rollkit.proto [hunk 4]
```diff
@@ -76,7 +76,6 @@ message Data {
 }
 
 message Block {	
-	Header header = 1;
+	SignedHeader signedHeader = 1;
```
Comment (tuxcanfly) [inline proto/rollkit/rollkit.proto:79]: nit: seems like we were using snake_case for `last_commit`?

File: types/block.go [hunk 5]
```diff
@@ -17,9 +17,8 @@ type Version struct {
 
 // Block defines the structure of Rollkit block.
 type Block struct {
-	Header     Header
-	Data       Data
-	LastCommit Commit
+	SignedHeader SignedHeader
```
Comment (tuxcanfly) [inline types/block.go:20]: could we embed `SignedHeader` here, so callers don't need to change?

**Review by tuxcanfly — APPROVED** [review]
Some nits, non blocking.

**Review by Manav-Aggarwal — APPROVED** [review]

**Review by gupadhyaya — APPROVED** [review]

**Review by Manav-Aggarwal — APPROVED** [review]

**Pull request merged**